tests/adapter/dbt/tests/adapter/caching/test_caching.py

```diff
@@ -0,0 +1,105 @@
+import pytest
+
+from dbt.tests.util import run_dbt
+
+model_sql = """
+{{
+    config(
+        materialized='table'
+    )
+}}
+select 1 as id
+"""
+
+another_schema_model_sql = """
+{{
+    config(
+        materialized='table',
+        schema='another_schema'
+    )
+}}
+select 1 as id
+"""
+
+
+class BaseCachingTest:
+    @pytest.fixture(scope="class")
+    def project_config_update(self):
+        return {
+            'config-version': 2,
+            'quoting': {
+                'identifier': False,
+                'schema': False,
+            }
+        }
+
+    def run_and_inspect_cache(self, project, run_args=None):
+        if not run_args:
```

Comment:
I think I did this so:
- `run_args` isn't a required argument whenever the `run_and_inspect_cache` method is called
- someone once told me that you should never set an arg default to a list, because [mutable default arguments are the source of all evil](https://florimond.dev/en/posts/2018/08/python-mutable-defaults-are-the-source-of-all-evil/)

Given that I'm calling `run_and_inspect_cache` in exactly two places, there isn't much need for a default. I can just pass the `run_args` in explicitly in both places.